src/components/TodoList/TodoList.tsx

```diff
@@ -1,100 +1,105 @@
+};
 
-    <tbody>
-      <tr data-cy="todo" className="">
-        <td className="is-vcentered">1</td>
-        <td className="is-vcentered" />
-        <td className="is-vcentered is-expanded">
-          <p className="has-text-danger">delectus aut autem</p>
-        </td>
-        <td className="has-text-right is-vcentered">
-          <button data-cy="selectButton" className="button" type="button">
-            <span className="icon">
-              <i className="far fa-eye" />
-            </span>
-          </button>
-        </td>
-      </tr>
-      <tr data-cy="todo" className="has-background-info-light">
-        <td className="is-vcentered">2</td>
-        <td className="is-vcentered" />
-        <td className="is-vcentered is-expanded">
-          <p className="has-text-danger">quis ut nam facilis et officia qui</p>
-        </td>
-        <td className="has-text-right is-vcentered">
-          <button data-cy="selectButton" className="button" type="button">
-            <span className="icon">
-              <i className="far fa-eye-slash" />
-            </span>
-          </button>
-        </td>
-      </tr>
+export const TodoList: React.FC<Props> = ({
+  todos,
+  filter,
+  title,
+  iconId,
+  setIsModalOpen,
+  setTodoCard,
+  setIconId,
+}) => {
+  let filtering;
 
-      <tr data-cy="todo" className="">
-        <td className="is-vcentered">1</td>
-        <td className="is-vcentered" />
-        <td className="is-vcentered is-expanded">
-          <p className="has-text-danger">delectus aut autem</p>
-        </td>
-        <td className="has-text-right is-vcentered">
-          <button data-cy="selectButton" className="button" type="button">
-            <span className="icon">
-              <i className="far fa-eye" />
-            </span>
-          </button>
-        </td>
-      </tr>
+  switch (filter) {
+    case FilteringType.active:
+      filtering = todos.filter((todo) => !todo.completed);
+      break;
+    case FilteringType.completed:
+      filtering = todos.filter((todo) => todo.completed);
+      break;
+    default: filtering = todos;
+  }
 
-      <tr data-cy="todo" className="">
-        <td className="is-vcentered">6</td>
-        <td className="is-vcentered" />
-        <td className="is-vcentered is-expanded">
-          <p className="has-text-danger">
-            qui ullam ratione quibusdam voluptatem quia omnis
-          </p>
-        </td>
-        <td className="has-text-right is-vcentered">
-          <button data-cy="selectButton" className="button" type="button">
-            <span className="icon">
-              <i className="far fa-eye" />
-            </span>
-          </button>
-        </td>
-      </tr>
+  const findMatches = filtering
+    .filter((todo) => todo.title.trim().toLowerCase()
+      .includes(title.trim().toLowerCase()));
+
+  const handleEyeButton = (todo: Todo) => {
+    setTodoCard(todo);
+    setIsModalOpen(true);
+    setIconId(todo.id);
+  };
```

Comment:
it's better to define such a function in the `App.tsx` and pass it as a prop here instead of passing 3 different "setState" functions

Reply:
can I leave this function if it passed 1 setState function?